src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -745,4 +757,12 @@ EvpKey translateKey(Key key, EvpKeyType keyType) throws InvalidKeyException {
       return (EvpKey) getKeyFactory(keyType).translateKey(key);
     }
   }
+
+  // In case the user does not register Ed25519 KeyFactories by ACCP, we still need one to be used
+  // internally.
+  private static class EdKeyFactory extends KeyFactory {
```

Comment:
I don't think this belongs inside the AmazonCorrettoCryptoProvider class, way too specific to a specific algorithm to be here. Maybe make it a standalone class or move it to the EvpKeyFactory class instead?

Reply:
KeyFactory is not intended to be instantiated explicitly: one should get an instance by calling the static factory methods KeyFactory::getInstance.
This usage is not common and `EdKeyFactory` is only used by [translateKey](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java#L741). I think it's better to be kept in this class.

Reply:
If we could do it all over again, I'd avoid having translateKey in this class as well. Way too much stuff inside here right now and it'd all benefit from a good refactor. Won't block the PR over it though.